processors/src/risk_engine.rs

```
@@ -112,58 +113,46 @@ impl RiskEngine {
-        if !self.user_id_for_this_handler(event.active_order_user_id)
-            && !self.user_id_for_this_handler(event.maker_user_id)
-        {
+        if !self.user_id_for_this_handler(event.maker_user_id) {
             return; // Not for this shard, skip
         }
 
         info!("[RiskEngine_{}] Handling event: {:?}", self.shard_id, event);
 
-        match event.event_type {
-            MatcherEventType::Trade => {
-                let spec = self.symbol_specs.get(&event.symbol_id).unwrap();
-
-                if let Some(maker_profile) = self.user_profiles.get_mut(&event.maker_user_id) {
-                    maker_profile.settle_trade(
-                        spec,
-                        event.price,
-                        event.size,
-                        if event.taker_action == Side::Ask {
-                            Side::Bid
-                        } else {
-                            Side::Ask
-                        },
-                    );
-                }
-                if let Some(taker_profile) = self.user_profiles.get_mut(&event.active_order_user_id)
-                {
-                    taker_profile.settle_trade(spec, event.price, event.size, event.taker_action);
+        // Determine maker side (opposite of taker side)
+        let maker_side = if taker_side == Side::Bid { Side::Ask } else { Side::Bid };
+        
+        if let Some(spec) = self.symbol_specs.get(&market_id) {
+            if let Some(maker_profile) = self.user_balances.get_mut(&event.maker_user_id) {
+                // For trade events, we need to settle the trade
+                // The maker's locked funds should be consumed
+                let trade_amount = event.price * event.size;
+                
+                if let Err(e) = maker_profile.consume_locked(event.maker_user_id, market_id, trade_amount) {
+                    warn!("[RiskEngine_{}] Failed to consume locked funds for maker {}: {:?}", 
```

The error is only logged but not propagated this could lead to silent failures